text/0083-sbt-standard.md

```diff
@@ -0,0 +1,255 @@
+- **created**: 09.08.2022
+
+# Summary
+
+Soul bound token (SBT) is a special kind of NFT which can be transferred only between its owner's accounts. For this, it stores immutable public key of the owner, and it is needed to send transfer from new address with signature in payload to change owner's address.
+
+# Motivation
+
+There is a useful type of token which allows to give social permissions/roles or certificates to some users. For example, it can be used by marketplaces to give discounts to owners of SBT, or by universities to give attestation certificates in SBT form. Mechanics with ownership proof allows to easily prove to any contract that you are an owner of some SBT.
+
+# Specification
+
+SBT implements [NFT standard interface](https://github.com/ton-blockchain/TIPs/issues/62) but `transfer` should always be rejected, `pull_ownership` is used instead.
+
+#### 1. `pull_ownership`
+
+TL-B schema of inbound message:
+```
+pull_ownership#03fdd6c9 query_id:uint64 signature:^(signature:(bits 512)) 
+sbt_nonce:uint64 new_owner:MsgAddress response_destination:MsgAddress 
+custom_payload:(Maybe ^Cell)
+```
+`query_id` -  arbitrary request number.
+
+`signature` - signature of the rest part of the message.
+
+`sbt_nonce` - nonce, required for protection of signature replay attacks.
+
+`new_owner` - address of the new owner of the SBT item, should be the same as sender's address.
+
+`response_destination` - address where a response with confirmation of a successful pull and the rest of the incoming message coins should be sent.
+
+`custom_payload` - optional custom data.
+
+**Should be rejected if:**
+1. `signature` verification failed.
+2. `sbt_nonce` not equals stored nonce.
+3. `new_owner` is not sender's address.
+4. there is not enough coins (with respect to NFT own storage fee guidelines).
+5. After processing the request, the contract must send at least in_msg_value - forward_amount - max_tx_gas_price to the response_destination address. If the contract cannot guarantee this, it must immediately stop executing the request and throw the error.
+
+**Otherwise should do:**
+1. Change current owner of SBT to `new_owner` address.
+2. Generate and store new random `nonce`.
+3. Send all excesses of incoming message coins to response_destination with the following layout:
+   
+   `excesses#d53276db query_id:uint64 = InternalMsgBody;`
+
+   query_id should be equal with request's `query_id`.
+
+#### 2. `prove_ownership`
+
+TL-B schema of inbound message:
+```
+prove_ownership#38061b82 query_id:uint64 dest:MsgAddress 
+data:^Cell with_content:bool
+```
+`query_id` -  arbitrary request number.
+
+`dest` -  address of the contract to which the ownership of SBT should be proven.
+
+`data` - arbitrary data required by target contract.
+
+`with_content` - if true, SBT's content cell will be included in message to contract.
+
+**Should be rejected if:**
+Sender address is not an owner address.
```

Comment:
Why?

Reply:
Good question! 

Yes, actually I see no reasons to restrict it, we can allow to trigger it to anyone to prove ownership with current owner's address.

Reply:
Updated implementation and description